### DISCUSSION TITLE
Preconfirmation Q&A

### DISCUSSION THREAD

**Discussion by GrapeBaBa** [ethstorage/pm#123] [Q&A]
**Po's questions**
1. Definition of preconfirmation? 
- Is there any other preconfirmation besides [based-preconfirmations](https://ethresear.ch/t/based-preconfirmations/17353)?
- What's the difference between designs by [justin](https://www.youtube.com/watch?v=2IK136vz-PM), Taiko and LimeChain
   - such as the difference in [preconfirmations flow](https://limechain.notion.site/RFC-Preconfirmations-Flow-Exploration-30fe218a0ea0443fb6bc213da969a47d)
2. How to commit precofirmation and slash preconfirmer?
- commit: Such as the [signed_commitment](https://github.com/LimeChain/based-preconfirmations-research/blob/main/docs/preconfirmations-for-vanilla-based-rollups.md#preconfirmation-commitment) method?
- slash?
3. How to enforce [inclusion and execution state preconfirmation](https://github.com/LimeChain/based-preconfirmations-research/blob/main/docs/preconfirmations-for-vanilla-based-rollups.md#execution-state-preconfirmation)?

**Comment by qizhou** [comment]
Looks like the links are missing from the original doc.

**Comment by qzhodl** [comment]
I have a question about how L2 will use the preconfirmation functionality. Since we can only know the block proposers for L1 for the next 32 to 64 slots, this gives us preconfirmation for only 6.4 to 12.8 minutes. However, if the batch frequency is 1 hour or longer (e.g., in cases where there’s low activity on smaller L2s), does preconfirmation still make sense in that scenario?

A potential solution is for multiple rollups to share the same blob, reducing the batching interval to two epochs or less.

**Reply by GrapeBaBa** [reply]
The requirement is to satisfy the user's wish that the transaction be included in the nearest slot and provide a confirmed commitment. This case does not seem to require preconfirmation. However, if L2 users have preconfirmation requirements, it means that L2 batch/block
will be submitted as soon as possible, which does conflict with the current optimism submission strategy.
